File: ApplicationCore/Entities/ItemVariantImage.cs
using ApplicationCore.Interfaces;
using System.IO;

namespace ApplicationCore.Entities
{
    /// <summary>
    /// Image of an Item Variant
    /// </summary>
    public class ItemVariantImage: Image<ItemVariant>, ShallowClonable<ItemVariantImage>
    {
        public ItemVariantImage(): base()
        {
        }

        public ItemVariantImage(string title, string contentType, int relatedId, string fullPath, Stream inputStream)
           : base(title, contentType, relatedId, null, inputStream)
        {
        }

        protected ItemVariantImage(ItemVariantImage itemVariantImage): base(itemVariantImage)
        {
        }

        public ItemVariantImage ShallowClone()
        {
            return new ItemVariantImage();
        }
    }
}

File: OspreyStore/ClientApp/src/app/item-variant-image/item-variant-image.ts
import { Image } from "./image";
import { ItemVariant } from "../item-variant/item-variant";

export class ItemVariantImage extends Image<ItemVariant>{
  shown: boolean;

  public constructor(init?: Partial<ItemVariantImage>) {
    super(init);
    Object.assign(this, init);
  }
}
